scipy/stats/stats.py

```diff
@@ -7469,6 +7470,214 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
     return BrunnerMunzelResult(wbfn, p)
 
 
+def _confint_lowerbound(n, quantile, confidence):
+    """Computes the lower bound for a one-sided confidence interval
+    for a given
+    - quantile (0<`quantile`<1)
+    - confidence level (0<`confidence`<1)
+    - number of samples `n`.
+
+    Returns the largest index of the sample being a valid lower bound,
+    or `None` if there are not enough samples to derive one.
+
+    Used by the public function confint_quantile().
+
+    .. versionadded:: 1.6.0
+    """
+    # compute all probabilities from the binomiale distribution for the quantile of interest
+    bd = binom(n, quantile)
+    ppm = [np.maximum(1-x, 0.0) for x in np.cumsum([bd.pmf(k) for k in range(n)])]
+
+    # search the index defining the lower-bound
+    if ppm[0] < confidence:
+        return None
+    else:
+        for k in range(n):
+            # search for first index reaching below the desired confidence
+            if ppm[k] < confidence:
```

Comment:
I wonder if `np.argmax(ppm < confidence)` might be a faster alternative to avoid raw Python looping?

Reply:
I suspect we can even write this as `binom.isf` or `binom.ppf` or directly `scipy.special.bdtrik`

Reply:
There are certainly multiple ways of implementing this. I haven't benchmarked this to see if there is a significant performance impact; I'll try doing this in the next few days

Reply:
I just tried it out: working with isf directly is about an order of magnitude faster. 
![image](https://user-images.githubusercontent.com/32954999/89767938-2e7fdf00-dafb-11ea-8499-e9b2f2e7431e.png)

I expect the speed-up gains to increase with `n` so I don't think it's worth investigating that further -> will update to use binom.isf instead.

Reply:
`binom.isf(confidence)` seems to return +1 what this function returns, not sure exactly why.

```
from scipy.stats import binom

def _confint_lowerbound(n, quantile, confidence):
    bd = binom(n, quantile)
    ppm = bd.pmf(0)
    sum_ppm = 1 - ppm
    if sum_ppm < confidence:
        print("What", sum_ppm, confidence)
        return None

    ratio = quantile / (1-quantile)
    for k in range(1, n):
        ppm *= (n-k+1) / k * ratio
        sum_ppm -= ppm
        if sum_ppm < confidence:
            return k - 1
    print(sum_ppm, ppm)
    return n

for c in [0.10, 0.27, 0.45, 0.89, 0.999, 0.9999]:
    n = 100
    quantile = 0.45
    test = _confint_lowerbound(n, quantile, c)
    test2 = binom(n, quantile).isf(c) - 1

    print(c, test, test2)
```
```
0.1 50 50.0
0.27 47 47.0
0.45 45 45.0
0.89 38 38.0
0.999 29 29.0
0.9999 26 26.0
```

Reply:
It is because `isf` returns the largest `k` such that `sf(k) < confidence`. So the lower bound of our confidence interval is `k-1`.